src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ResourceExtensions.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="ResourceExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenTelemetry.Resources;
+using OtlpCommon = Opentelemetry.Proto.Common.V1;
+using OtlpResource = Opentelemetry.Proto.Resource.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation
+{
+    internal static class ResourceExtensions
+    {
+        public static OtlpResource.Resource ToOtlpResource(this Resource resource)
+        {
+            var processResource = new OtlpResource.Resource();
+
+            foreach (KeyValuePair<string, object> attribute in resource.Attributes)
+            {
+                var oltpAttribute = attribute.ToOtlpAttribute();
+                if (oltpAttribute != null)
+                {
+                    processResource.Attributes.Add(oltpAttribute);
+                }
+            }
+
+            if (!processResource.Attributes.Any(kvp => kvp.Key == ResourceSemanticConventions.AttributeServiceName))
+            {
+                var serviceName = (string)ResourceBuilder.CreateDefault().Build().Attributes.Where(
+                    kvp => kvp.Key == ResourceSemanticConventions.AttributeServiceName).FirstOrDefault().Value;
```

Comment:
In moving this code, the logic here changed a little. When this code was on the `OtlpTraceExporter` class it was:

```csharp
var serviceName = (string)this.ParentProvider.GetDefaultResource().Attributes.Where(
                    kvp => kvp.Key == ResourceSemanticConventions.AttributeServiceName).FirstOrDefault().Value;
```

Reply:
I think we could simplify the linq a bit (combine Where + FirstOrDefault) as: `(string)this.ParentProvider.GetDefaultResource().Attributes.FirstOrDefault(kvp => kvp.Key == ResourceSemanticConventions.AttributeServiceName).Value`